supervision/annotators/core.py

```diff
@@ -74,20 +173,604 @@ def annotate(self, scene: np.ndarray, detections: Detections) -> np.ndarray:
 
 
 class LabelAnnotator(BaseAnnotator):
-    def annotate(self, scene: np.ndarray, detections: Detections) -> np.ndarray:
-        pass
+    """
+    A class for putting text on an image using provided detections.
 
+    Attributes:
+        color (Union[Color, ColorPalette]): The color to text on the image, can be a single color or a color palette
+    """
 
-class PillowLabelAnnotator(BaseAnnotator):
-    def annotate(self, scene: np.ndarray, detections: Detections) -> np.ndarray:
-        pass
+    def __init__(
+        self,
+        color: Union[Color, ColorPalette] = ColorPalette.default(),
+        thickness: int = 2,
+        text_color: Color = Color.black(),
+        text_scale: float = 0.5,
+        text_thickness: int = 1,
+        text_padding: int = 10,
+        color_by_track: bool = False,
```

Comment:
Let's add `label_formatter: Callable[[Detections], List[str]]` argument.

The default `label_formatter` could be something like this.

```python
def default_label_formatter(detections: Detections) -> List[str]:
    return [
        str(class_id) 
        for class_id 
        in detections.class_id
    ]
```

If you want to use class mapping in your `label_formatter` you could do it like this:

```python
def build_label_formatter(classes: List[str]) -> Callable[[Detections], List[str]]:
    def default_label_formatter(detections: Detections) -> List[str]:
        return [
            classes[class_id] 
            for class_id 
            in detections.class_id
        ]
    return default_label_formatter
```

Reply:
Hi @SkalskiP :wave: , Thanks for the idea. I think it is too much for just to create `label`. This type of callback is not a good idea. It makes complicated everything. Instead, what do you think of this:

```python

class LabelAnnotator(BaseAnnotator):
	def __init__(..., labelmaps: Optionl[Dict[int, str]]=None):
		self.labelmaps = labelmap
		
	def annotate(self,  scene: np.ndarray,  detections: Detections):
	    for det in detections:
			if self.labelmaps:
		       	label=self.labelmaps[classid]
		    if not self.labelmaps:
			    labels = f"{classid}"				  

```
An example of labelmaps would be like:

```
labelmaps = {0: 'person', 1: '...', 2: '...'}
```
My concern is that if user has to write callback function then mostly they are not going to use this from `supervision` but instead they will use their own.

Reply:
Hi @hardikdava 👋🏻, label map is, of course, a lot simpler, but a mall problem I see is not flexible :/ It doesn't allow you to build complicated labels.

How about adding both `classes: Optional[List[str]], label_formatter: Optional[Callable[[Detections], List[str]]]`?

Reply:
@SkalskiP, That seems fine to me. User can have flexibility more when they want to use some other format.

Reply:
@hardikdava exactly! There will likely be multiple people asking about label customization. And I want to support them as well :)

Reply:
It is done. Ready for test.

Reply:
Awesome! 🔥 Let me test it. We will probably need to add unit tests for `get_anchor_coordinates`. Would you like me to take care of that?

Reply:
Okay. I will try. What is command line for running tests? basic pytest or something special?